crates/sncast/src/starknet_commands/account/add.rs

```diff
@@ -23,6 +24,10 @@ pub struct Add {
     #[clap(short, long, requires = "private_key_input")]
     pub address: FieldElement,
 
+    /// Type of the account
+    #[clap(short = 't', long = "type")]
+    pub account_type: AccountType,
```

Comment:
I am not sure if `type` is the correct word. `provider` maybe would be better? Or implementation?

Also, if i use help does ValueEnum list the options? if not we should document it.

Reply:
Starkli refers to those as `variants` https://book.starkli.rs/accounts imo it is better than `type` but still a bit off. 

If `provider` is taken I would fall back to `implementation`. `Type` seems off to me as it suggests set of accounts characterised by some properties. In our case each of those is a singleton, there are no set of properites, whose implementations implement the same interface.

Reply:
> Starkli refers to those as variants

Only within the documentation, in account files it is also a `type`.

> Type seems off to me as it suggests set of accounts characterised by some properties

Actually, I think it can be interpreted this way if we think about different class hashes as instances of the same account type. However, I would think about it in a different way. There is an account at the `sncast` level that is characterised by the type. The cast interface is the same for every type, but under the hood, different types have different implementations, especially for constructor and signature logic.

Reply:
Account abstraction is a specific interface and those are different implementations, it is much better word and I don't see any valid argument why we shouldn't use it. 
We even use the word `implementation` in the explanation, why just don't name the set od implementations `implementation` instead of naming a set of implementations `type`. 
```
 Possible values:
          - oz:     OpenZeppelin account implementation
```


> Actually, I think it can be interpreted this way if we think about different class hashes as instances of the same account type. However, I would think about it in a different way. There is an account at the sncast level that is characterised by the type. The cast interface is the same for every type, but under the hood, different types have different implementations, especially for constructor and signature logic.

I don't understand this explanation, sorry 😞

Reply:
> Account abstraction is a specific interface and those are different implementations

technically, and correct me if I'm wrong, an account may support a different signer (eg braavos), or have a unique features in comparison to other account types/implementations, so it then technically alters an interface, right? point is, while the said contract must fulfill specific requirements in order to be considered an account, it may also augument them. In that case I would argue if a word 'implementation' is better that a 'type' (although generally I think both of those words are imperfect in our case).

In my opinion a big argument to leave this as 'type' is that (as Darek said) starkli already uses this nomenglature in their codebase, and I don't think having a 'type' field in starkli-style account files and something different like 'implementation' in legacy-starknet-style account files we utilize by default is a wise thing to do - it introduces unnecessary complexity to our code (handling two names for the same thing - remember we cannot introduce this breaking change to starkli accounts) and is just plain pointless.

The question then arises - if we use 'type' in the files holding accounts (which I very strongly suggest), shouldn't we use the same word in docs in order to not introduce more confusion among users? I'd say we should, but I see your point that the word does not properly describe differences between oz/argent/braavos. Perhaps there is a better word to describe it altogether, and I'm keen to hear your ideas lads, I'm just not sure.

Reply:
> In my opinion a big argument to leave this as 'type' is that (as Darek said) starkli already uses this nomenglature in their codebase

In my opinion it is not an argument pro, and even should be considered slightly against, to correct our initial bias. We should pick the name that describes the property the best, we should not follow every mistake that starkli does, foundry has more stars on github. We should put an effort to build the best possible standard.

Reply:
> an account may support a different signer (eg braavos), or have a unique features in comparison to other account types/implementation

Agreed there could be multiple implementations of an account by a single provider with different features. If other providers would follow the same feature sets distinction, then this could eventually yield an account types. Though still I think we should allow users to select an `implementation` here as user selects specific code.

Reply:
> if we use 'type' in the files holding accounts (which I very strongly suggest), shouldn't we use the same word in docs in order to not introduce more confusion among users? I'd say we should

I think we should use the word implementation everywhere, because that's what it is. We should put an effort to correct the mistakes if we observe them, not just follow "others do that" or "we did it already somewhere else"

Reply:
> not just follow "others do that" or "we did it already somewhere else"

this is not the case at all - this is a case of "We must support starkli accounts, and they are using 'type' fields, so in order to not break things we should use the same". I don't think there is sensible effort we could put in to "correct the mistake", because we are depedend on starkli with this one - if we deviate it is a breaking change that makes starlki accounts unusable, and as I said - there is no point in my opinion to have "type" in one file and "implementation" in other, this just make our code more complex and can lead to issues.

Reply:
> because we are depedend on starkli with this one

I am not sure why this is a breaking change, changing the flag does not have to modify the fields in the account file. We just offer a compatibility layer over starkli, we don't have to do the same.

Reply:
Discussed offline, let's leave the name `type`